### ISSUE TITLE
Bug: [member-ordering] Not Detecting Order For Read-Only Fields Correctly

### ISSUE THREAD

**Issue by tyrone-gallardo-oobnow** [typescript-eslint/typescript-eslint#9098]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.3&fileType=.tsx&code=KYDwDg9gTgLgBAYwDYEMDOa4GFUbgbwFgAoOOKYFAEwgDskBPOAJQHk4BeOAchQCME3ANwkSZAGKce-QSOIBfEkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AW0XYCNFpaA9tAAmfZgHN0YANrhsOPtCGQANHPlZ5WyKIBmAQwT4pmrds48%2BAFRIopss2ZiJ9wgU3iFauyonjDVdUccYlhuRjJaUTIhfXxEYW9ff0Dg%2BUhQ8MpI6Nj4xOREYzU07AyleLJ8qMQY6DiEpL8AktKoYgra6tz66sLioODyygA3Bpq6hsSfZtS24bH4ibzG-rnSjLCI2mZkfH0mMkQmlNaNjoFK6t39w%2BOZ08HHBfGbg6OTlqenTO2Ypj2cCqyjOaQcbRCW2ytHERXW81%2B0LW3y0AF1QcFwRDypcuo1YQMISFOlVVnCUfJ0RTsFiEVRFscCfDzvTxsiIVTsYjIm87rROPgABYCL5cknXAG3D4C4WiumjV6S97HGUi5nPbk7JV81Vylm40mJXnSoqy9U-VlLY0q01q6mc%2BZCUTQKSQfTwYiC-S8fDZd2eSAogC%2BTxDlLkIaDQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
Check playground please.
```

### ESLint Config

```javascript
import globals from "globals";
import pluginJs from "@eslint/js";
import tseslint from "typescript-eslint";

export default [
  { languageOptions: { globals: globals.browser } },
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,
  {
    rules: {
      "@typescript-eslint/member-ordering": [
        "error",
        {
          default: {
            memberTypes: [
              "readonly-field",
              "public-decorated-field",
              "public-decorated-set",
              "protected-decorated-field",
              "protected-decorated-set",
              "private-decorated-field",
              "private-decorated-set",
              "public-instance-field",
              "protected-instance-field",
              "private-instance-field",
            ],
            order: "alphabetically",
          },
        },
      ],
    },
  },
];
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expect tslint to not lift an error, since I configured readonly fields to go before instance fields

### Actual Result

tslint throws error `Member F should be declared before member RO. 4:3 - 4:13`

**Comment by bradzacher** [comment]
This may be a little counter-intuitive but it is working as intended.

The rule looks for "selectors" in a pre-defined order and it uses the specificity of the selector to determine what matches first.

For a public readonly field that is not static, the most specific selector for it is `public-instance-readonly-field` and the least specific is `readonly-field`.

In your example you have the least specific selector `readonly-field` and a more specific selector `public-instance-field`. So the member matches the latter selector before it matches the former. Meaning that both members match the same selector. Meaning the rule then uses your `order` config (alphabetical).

Given that you have granularly defined very specific selectors of `*-instance-field` et al - you also need to granularly define selectors for `*-readonly-field` so that the rule correctly matches. For example if you define `public-instance-readonly-field` instead of `readonly-field` then the rule works as intended [playground](https://typescript-eslint.io/play/#ts=5.4.3&fileType=.tsx&code=KYDwDg9gTgLgBAYwDYEMDOa4GFUbgbwFgAoOOKYFAEwgDskBPOAJQHk4BeOAchQCME3ANwkSZAGKce-QSOIBfEkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AW0XYCNFpaA9tAAmfZgHN0YANrhsOPtCGQANHPlZ5WyKIBmAQwT4pmrds48%2BAFRIopss46jFY3RmVrNk%2BfUzKJaaER9YQEmeEJaXUpEeGFVdSdsSBc3Sg8vHz8A6Nj4xKcAXTUk5KFRaClIfXhiAAt9Xnx0mojIAuwAXw7u%2BUK5bu6gA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)
